extensions/hiit/src/components/intervalForm.tsx

```diff
@@ -0,0 +1,91 @@
+import { Action, ActionPanel, Form } from "@raycast/api";
+import { IntervalFormValues, Item } from "../types";
+import { nanoid } from "nanoid";
+import { calculateInterval, requiresNumberGreaterThan, secondsToTime, setsToSeconds } from "../utils";
+import { FormValidation, useForm } from "@raycast/utils";
+
+export function IntervalForm(props: { item?: Item; onSave: (item: Item) => void }) {
+  const { handleSubmit, itemProps, setValue } = useForm<IntervalFormValues>({
+    async onSubmit(values) {
+      const sets = parseInt(values.sets);
+      const warmup = values.warmup ? parseInt(values.warmup) : 0;
+      const cooldown = values.cooldown ? parseInt(values.cooldown) : 0;
+      const high = parseInt(values.high);
+      const low = parseInt(values.low);
+
+      const interval: Item = {
+        title: values.title,
+        subtitle: values.subtitle,
+        interval: {
+          warmup: warmup,
+          cooldown: cooldown,
+          high: high,
+          low: low,
+          sets: sets,
+          intervals: calculateInterval(sets, warmup, cooldown),
+          totalTime: setsToSeconds(sets, high, low, warmup, cooldown),
+        },
+      };
+
+      props.onSave(props?.item ? { ...interval, id: props.item.id } : { ...interval, id: nanoid() });
+    },
+    validation: {
+      title: FormValidation.Required,
+      warmup: (value) => {
+        return requiresNumberGreaterThan(value, 1);
+      },
+      cooldown: (value) => {
+        return requiresNumberGreaterThan(value, 1);
+      },
+      high: (value) => {
+        return requiresNumberGreaterThan(value, 0);
+      },
+      low: (value) => {
+        return requiresNumberGreaterThan(value, 0);
+      },
+      sets: (value) => {
+        return requiresNumberGreaterThan(value, 1);
+      },
+    },
+    initialValues: {
+      title: props.item?.title,
+      subtitle: props.item?.subtitle,
+      warmup: props.item?.interval.warmup ? props.item.interval.warmup.toString() : "",
+      cooldown: props.item?.interval.cooldown ? props.item.interval.cooldown.toString() : "",
+      high: props.item?.interval.high ? props.item.interval.high.toString() : "",
+      low: props.item?.interval.low ? props.item.interval.low.toString() : "",
+      sets: props.item?.interval.sets ? props.item.interval.sets.toString() : "",
+    },
+  });
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title="Submit" onSubmit={handleSubmit} />
+        </ActionPanel>
+      }
+    >
+      <Form.TextField title="Title" placeholder="My Awesome Interval" {...itemProps.title} />
```

Comment:
tiny nit, just change the casing for consistency 
```suggestion
      <Form.TextField title="Title" placeholder="My awesome interval" {...itemProps.title} />
```